### PR TITLE
styling fixed and cancel button added in teststeps badge in testcases

### PR DESCRIPTION
**Pull request by tanmaythole**

### CODE REVIEW

File: src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import React, { useState } from "react";
-import PropTypes from "prop-types";
 import { Form } from "react-bootstrap";
+import { X } from "react-bootstrap-icons";
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx]: use as for this and rename it to `cancelIcon`

File: src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx [hunk 2]
```diff
@@ -77,30 +86,32 @@ const AddNewTestcase = (props) => {
...
-              {selectedTeststeps?.map((step, index) => {
-                return (
-                  <BadgeComponent
-                    key={index}
-                    bg="secondary"
-                    label={step.name}
-                    className="py-2 px-3 me-2"
+            <div className="background-secondary w-100 py-3 px-3 rounded d-flex">
+              {teststeps?.map(({ name: stepName }, index) => (
+                <div
```
Comment (ChinmaySonawane) [inline src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx]: can we write separate component in this same file

File: src/containers/ProjectContainer/TestcaseContainer/index.jsx [hunk 3]
```diff
@@ -177,10 +178,23 @@ const TestcaseContainer = (props) => {
...
+        testdata: teststep.selected_testdata,
+      })),
+    };
+
     if (cat === "add") {
-      dispatch(addTestcasesRequest(testcaseFormData));
+      dispatch(addTestcasesRequest(formDataToSubmit));
     } else {
-      dispatch(editTestcasesRequest(testcaseFormData));
+      dispatch(editTestcasesRequest({ id, ...formDataToSubmit }));
```
Comment (ChinmaySonawane) [inline src/containers/ProjectContainer/TestcaseContainer/index.jsx]: change the order for id...
it should be 
{ ...formDataToSubmit, id }
as if incase formDataToSubmit had id in it ,then our id should not be override with formDataToSubmit's id

File: src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx (hunk 2 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx]: 👍🏻

File: src/containers/ProjectContainer/TestcaseContainer/index.jsx (hunk 3 above)
Comment (tanmaythole) [reply src/containers/ProjectContainer/TestcaseContainer/index.jsx]: 👍🏻

File: src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx (hunk 1 above)
Comment (tanmaythole) [reply src/Components/ProjectsComponent/TestcaseComponents/AddNewTestcase.jsx]: 👍🏻

**Review by ChinmaySonawane — APPROVED** [review]

**Pull request merged**